vars/manifest.yml

```diff
@@ -2,37 +2,46 @@
 upstream_repos:
   openstack-helm-infra:
     src: https://git.openstack.org/openstack/openstack-helm-infra
-    version: master
+    # Commit on April 24, 2019
+    version: 0ed4f0de5ed7b0eb7bb6cb6f819ad48e3fd71a23
```

Comment:
I think we need to spend a little more thought on this and add one layer of indirection. what I can imagne is a list of repo/githashes under a specific tag (like e.g. ccp_preview_m1) and only if some magic flag is set (env) then those git hashes are used instead of master. 

we could set that flag by default globlaly somewhere, but imho there has to be a convenient way to set everything back to "latest" for continuous integration and development purposes. maybe we connect it to the already existing developer mode flag for now.

Reply:
Something like we have already for git project. `latest`, `stable-X` or a specific commit.

Reply:
I would also do a developer_mode -> latest + fetch the upstream patches, if not, use the specific sha. I agree a different flag could be used too, to avoid mixing things up, and still keep the consistent interface. Because everything is cloned with ansible, we could have conditionals and ternaries in the "version" line.